### DISCUSSION TITLE
How to direct to Auth0 login screen

### DISCUSSION THREAD

**Discussion by vnugent** [nextauthjs/next-auth#4790] [Help]
I have a custom login page at `/auth/signin` with just a button that calls `signIn('auth0')`.  When I click on the button, it takes me to Auth0 signin page with a popup asking for username and password.

How do I protect an API endpoint and direct users directly to Auth0 sign in page from the backend?  Basically, I'm looking for a sever-side version of `signin('auth0')`.   

```
import { getSession } from 'next-auth/react'
import { NextApiHandler } from 'next'

const withAuth = (handler: NextApiHandler): NextApiHandler => {
  return async (req, res) => {
    const session = await getSession({ req })
    if (session != null) {
      await handler(req, res)
    } else {
      res.status(401).redirect(307, '/auth/signin')
    }
  }
}

export default withAuth

```

**Comment by vnugent** [comment]
Anyone?

**Comment by paul-vd** [comment]
I'm wondering the same thing

**Reply by paul-vd** [reply]
I would expect it should be possible using the [`pages`](https://next-auth.js.org/configuration/pages) option, but not sure if it would be possible to get the URL for auth0

**Reply by paul-vd** [reply]
I don't think it's possible "yet", if you look at the signIn function, it calls the auth api https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/react/index.tsx#L252-L264 and then from that response it has the url which is used.
